frontend/src/Quiz/QuizAssign.vue

```diff
@@ -0,0 +1,318 @@
+                      :disabled="!clazz.assigned"
+                      placeholder="Deadline"
+                    />
+                    <button
+                      class="btn btn-sm btn-secondary"
+                      title="Set deadline to all assigned classes"
+                      :disabled="!clazz.deadline"
+                      @click="setToAllClassesByProperty(clazz, 'deadline')"
+                    >
+                      <span class="iconify" data-icon="mdi:content-duplicate"></span>
+                    </button>
+                  </div>
+                  <div class="col-lg d-flex justify-content-start">
+                    <input
+                      v-model="clazz.duration"
+                      placeholder="Duration (minutes)"
+                      class="form-control"
+                      type="number"
+                      min="1"
+                      step="1"
+                      :disabled="!clazz.deadline || !clazz.assigned"
+                    />
+                    <button
+                      class="btn btn-sm btn-secondary"
+                      title="Set duration to all assigned classes"
+                      :disabled="!clazz.duration"
+                      @click="setToAllClassesByProperty(clazz, 'duration')"
+                    >
+                      <span class="iconify" data-icon="mdi:content-duplicate"></span>
+                    </button>
+                  </div>
+                </div>
+              </td>
+              <td>
+                <div class="form-check d-flex p-0">
+                  <input
+                    :id="'publish-results-input-' + clazz.id"
+                    v-model="clazz.publish_results"
+                    class="form-check-input"
+                    type="checkbox"
+                    :disabled="isMissingPropertyToAssign(clazz)"
+                    :checked="clazz.publish_results"
+                  />
+                  <label :for="'publish-results-input-' + clazz.id" class="form-check-label"
+                    >&nbsp;Publish results</label
+                  >
+                </div>
+              </td>
+              <td>
+                <button
+                  class="btn btn-secondary"
+                  title="Set values to all assigned classes"
+                  :disabled="isMissingPropertyToAssign(clazz)"
+                  @click="setToAllClasses(clazz)"
+                >
+                  <span class="iconify" data-icon="mdi:content-duplicate"></span>
+                </button>
+              </td>
+              <td>
+                <button
+                  class="btn-close"
+                  aria-label="Unassign class"
+                  :disabled="!clazz.deletable"
+                  @click="deleteAssignmentValues(clazz)"
+                ></button>
+              </td>
+            </tr>
+          </tbody>
+        </table>
+      </div>
+    </div>
+  </div>
+</template>
+
+<style scoped>
+#assign:deep(td) {
```

Comment:
Why is `:deep` here? That is only required if you wanted to style `td` inside a nested Vue component, but I don't see any of that in this template, it's just HTML.